### ISSUE TITLE
Connect with Multiple Product Catalogs

### ISSUE THREAD

**Issue by coccoinomane** [facebookincubator/facebook-for-woocommerce#533]
Hello!

**TL;DR**: Is it possible to connect the feed generated by Facebook For WooCommerce to two different Product Catalogs in Facebook?

**Context**

We have been using Facebook For WooCommerce for a while on three e-commerce, and so far it has worked great.

Now we have the need of tagging products from the same e-commerce in two different Instagram accounts.

Facebook support has told us that the only way to do so is to create two different Product Catalogs in Facebook, and to connect both of them to the same feed.

Can we achieve this given that we use Facebook For WooCommerce on the e-commerce?

Thank you,
Guido

**Comment by coccoinomane** [comment]
Feature Request

**Comment by dmitridr** [comment]
Hello,

Thanks for your interest in our plugin, glad you like it! Please consider giving us [a review on woocommerce](https://woocommerce.com/products/facebook/).

This is an interesting problem that you are facing. 

After some thought... I think there may be a way for you to get the result you want with some small work on your part of editing of the plugin, just by adding one line! 

Here is my idea:
(1) If you have a Catalog you want to sync to, create an empty feed in that catalog (or feed with 1 product). Get the ID of this feed (it should be in the url).
(2) Use the Wordpress plugin editor to insert a new line in the plugin [right here](https://github.com/facebookincubator/facebook-for-woocommerce/blob/master/includes/fbproductfeed.php#L65). The line should be:

```
$this->upload_id = $this->create_upload(<FEED_ID>);
```
Where `<FEED_ID>` is the ID you got in (1).
Note the line below it, by making this change you are uploading products into two FB feeds.

(3) Next time you `Resync` products using the provided button, products will be uploaded to both catalogs. Note that you will have to keep `Resync`-ing periodically to keep the second catalog up to date, that's the disadvantage of this workaround.

If this works for you, we can potentially fully support your case with the feature requested in #542, by creating a mechanism to schedule periodic resyncing.

**Comment by dmitridr** [comment]
One additional note : The 2nd catalog you are using should be editable by the user who set up the plugin, otherwise the feed upload would fail due to permission problems.

**Comment by coccoinomane** [comment]
Thank you for the nice solution dmitridr!
Alas, I've tried it but it doesn't seem to work.

1. I've added the line where you told me (see screenshot) using an empty catalog (ID 1577052032424571) and forced a re-sync.
1. The re-sync starts but it soon freezes with the message "Syncing... Keep this browser open Until sync is complete". In `debug.log` I do see the expected "Pushing post to queue:" messages, but the sync does not proceed.
1. If I refresh the page and click again on force re-sync, the file `debug.log` shows "Not syncing, sync in progress". Then, the sync proceeds correctly and all products are synced (Background sync complete!).
1. When I check the new catalog on Facebook (https://www.facebook.com/products/catalogs/1577052032424571/products) I still see no products.

I am sure to have the correct permissions on the catalogs, as I created both of them.
Maybe I should add something else in the code?

Thank you,
Guido

PS: Just an idea for you: have you considered putting `do_action()` in `sync_all_products_using_feed()`? Something like `do_action( 'facebook_for_woocommerce_before_sync', $this );` would allow both internale devs and plugin/theme devs to hook into the code without having to change anything in the core codebase.

**Comment by dmitridr** [comment]
Sorry for the delay in getting back to you Guido.

It's unfortunate that the solution did not work. For the code replacement you will need to use the ID of the feed associated with the catalog, and not the catalog itself. If it's an empty catalog you can create a feed for it in catalog manager. I'm not sure if you did that... as I do not see the screenshot you referenced, I guess it did not upload?

I think it definitely makes sense to add more filters such as the one you suggest. We are happy to add them, but we are swamped right now, please feel free to send us a pull request!

I will think more about how we can support this case, perhaps we can allow an advanced option to sync to a different catalog/feed.

**Comment by coccoinomane** [comment]
Thanks @dmitridr , and sorry for the missing screenshot!
I probably used the catalog ID instead of the feed ID, I will

**Comment by lgfogle** [comment]
Hey there! 👋

Lindsey here, from the SkyVerge product team! We recently took over development and support for the Facebook for WooCommerce extension.

In our recent 1.10 releases, we've fixed many issues impacting the store connection, pixel tracking, and product sync. Could you please update to the latest version of Facebook for WooCommerce and see if that resolves this issue?

If not, could you please [contact our support team](https://woocommerce.com/my-account/create-a-ticket/) or open a new issue here? We'd be happy to help troubleshoot further with you.

Thank you!
